autonomi/src/client/config.rs

```diff
@@ -0,0 +1,315 @@
+    pub(crate) read_retry_strategy: RetryStrategy,
+}
+
+#[derive(Debug, Clone)]
+pub struct GraphOperationConfig {
+    /// The retry strategy to use if we fail to store a graph entry. Every write will also verify that the entry
+    /// is stored on the network by fetching it back.
+    ///
+    /// Use the `verification_quorum` and `verification_retry_strategy` to configure the verification operation.
+    pub(crate) write_retry_strategy: RetryStrategy,
+    /// The number of entries to wait for before considering the read after write operation successful.
+    pub(crate) verification_quorum: ResponseQuorum,
+    /// The retry strategy to use if the read after write operation fails.
+    pub(crate) verification_retry_strategy: RetryStrategy,
+    /// The number of entries to wait for before considering the read operation successful.
+    pub(crate) read_quorum: ResponseQuorum,
+    /// The retry strategy to use if the read operation fails.
+    pub(crate) read_retry_strategy: RetryStrategy,
+}
+
+#[derive(Debug, Clone)]
+pub struct PointerOperationConfig {
+    /// The retry strategy to use if we fail to store a pointer. Every write will also verify that the pointer
+    /// is stored on the network by fetching it back.
+    ///
+    /// Use the `verification_quorum` and `verification_retry_strategy` to configure the verification operation.
+    pub(crate) write_retry_strategy: RetryStrategy,
+    /// The number of pointers to wait for before considering the read after write operation successful.
+    pub(crate) verification_quorum: ResponseQuorum,
+    /// The retry strategy to use if the read after write operation fails.
+    pub(crate) verification_retry_strategy: RetryStrategy,
+    /// The number of pointers to wait for before considering the read operation successful.
+    pub(crate) read_quorum: ResponseQuorum,
+    /// The retry strategy to use if the read operation fails.
+    pub(crate) read_retry_strategy: RetryStrategy,
+}
+
+#[derive(Debug, Clone)]
+pub struct ScratchpadOperationConfig {
+    /// The retry strategy to use if we fail to store a scratchpad. Every write will also verify that the scratchpad
+    /// is stored on the network by fetching it back.
+    ///
+    /// Use the `verification_quorum` and `verification_retry_strategy` to configure the verification operation.
+    pub(crate) write_retry_strategy: RetryStrategy,
+    /// The number of scratchpads to wait for before considering the read after write operation successful.
+    pub(crate) verification_quorum: ResponseQuorum,
+    /// The retry strategy to use if the read after write operation fails.
+    pub(crate) verification_retry_strategy: RetryStrategy,
+    /// The number of scratchpads to wait for before considering the read operation successful.
+    pub(crate) read_quorum: ResponseQuorum,
+    /// The retry strategy to use if the read operation fails.
+    pub(crate) read_retry_strategy: RetryStrategy,
+}
```

Comment:
Those are exactly the same please make this one type. If in the future we need to make them different, we will. But for now, please let's keep this simple.

Reply:
Updated. But this would mean that we lose the ability to set the cfgs in one place. We reuse the configs at some places and those might be prone to errors.

Reply:
Ok will address that in a follow up then